packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -193,13 +193,25 @@ export function createObjectProxy<TSchema extends FlexObjectNodeSchema>(
 
 			return true;
 		},
+		// The "has" trap is called when a user checks for a property with "in" (e.g. `if ("z" in point) {...}`).
 		has: (target, key) => {
-			return (
+			if (
 				schema.objectNodeFields.has(key as FieldKey) ||
 				(allowAdditionalProperties ? Reflect.has(target, key) : false)
-			);
+			) {
+				// Filter out properties which are not own-enumerable, for example, the keys of optional fields that are in the schema, but whose value is currently `undefined`.
+				// This filtering is delegated to the "getOwnPropertyDescriptor" trap below, which also handles the filtering of keys in other scenarios (see the "ownKeys" trap below).
+				return Object.getOwnPropertyDescriptor(proxy, key)?.enumerable ?? false;
+			}
+
+			return false;
 		},
+		// The "ownKeys" trap is called by Object.keys() and for-in loops.
 		ownKeys: (target) => {
+			// Javascript calls the "getOwnPropertyDescriptor" for each key when iterating over an object via `Object.keys()`.
```

Comment:
If you look at the intelisense for this ownKeys member, you get the docs "A trap for Reflect.ownKeys()."

P{roxies have to implement the behavior required by A trap for Reflect.ownKeys(), not just something that happens to make Object.keys work as expected.

Object.keys is different from Reflet.ownKeys, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/ownKeys Note that "Reflect.ownKeys() provides the reflective semantic of retrieving all property keys of an object. It is the only way to get all own properties – enumerable and not enumerable, strings and symbols — in one call, without extra filtering logic."

We should not include empty fields here! It seems that bug has not been fixed.

Reply:
Discussed offline, thank you.